lib/Dialect/TritonIntelGPU/Transforms/PrefetchBlock.cpp

```diff
@@ -0,0 +1,269 @@
+//===-------------- PrefetchBlock.cpp -  ------------------------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+/// \file
+/// This file implements a pass to add prefetch mechanism for target that
+/// supports memory prefetch.
+/// This pass match certain pattern of scf.loop with tt.load and then add
+/// prefetch both in the loop preheader(3 stages in advance) and loop body.
+/// This pass only support cases with block pointer.
+/// This pass should be run after triton-to-tritongpu-warp.
+/// This pass also add layout Attr to newly created ops.
+///
+/// before this pass:
+///  scf.for
+///    tt.load
+///    tt.advance
+///
+/// after this pass:
+/// tt.make_tensor_ptr
+/// tt.prefetch
+/// tt.advance
+///  scf.for
+///    tt.load
+///    tt.advance
+///    tt.prefetch
+///    tt.advance
+//===----------------------------------------------------------------------===//
+
+#include "mlir/IR/BuiltinAttributes.h"
+#include "mlir/IR/IRMapping.h"
+#include "mlir/IR/Matchers.h"
+#include "mlir/IR/PatternMatch.h"
+#include "mlir/Pass/Pass.h"
+#include "mlir/Support/LogicalResult.h"
+
+#include "triton/Dialect/Triton/IR/Dialect.h"
+#include "triton/Dialect/TritonGPU/IR/Dialect.h"
+#include "triton/Dialect/TritonIntelGPU/IR/Dialect.h"
+#include "triton/Dialect/TritonIntelGPU/Transforms/Passes.h"
+
+#include <memory>
+
+namespace mlir {
+#define GEN_PASS_CLASSES
+#include "triton/Dialect/TritonIntelGPU/Transforms/Passes.h.inc"
+} // namespace mlir
+
+using namespace mlir;
+namespace tt = mlir::triton;
+namespace ttg = mlir::triton::gpu;
+namespace ttig = mlir::triton::gpu::intel;
```

Comment:
a reviewer suggested this in another PR
```suggestion
namespace ttgi = mlir::triton::gpu::intel;
```

Reply:
but the upsteam triton using ttng for nvidia_gpu, shouldn't we follow the same?

Reply:
I am fine either way, but prefer to consistence for all our `mlir::triton::gpu::intel`.
@etiotto are you ok with `ttig`?

Reply:
ah, we make our namespace "triton::gpu::intel" instead of "triton::intel_gpu"
so ttgi makes sense.
but later we can discuss and choose to use what namespace style.